packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -2616,6 +2599,42 @@ export class ProjectService extends BaseService {
             this.lightdashConfig.query.timezone || 'UTC',
         );
 
+        const queryHashKey = metricQuery.timezone
+            ? `${projectUuid}.${query}.${metricQuery.timezone}`
+            : `${projectUuid}.${query}`;
+        const queryHash = crypto
+            .createHash('sha256')
+            .update(queryHashKey)
+            .digest('hex');
+
+        const isCacheEnabled =
+            this.lightdashConfig.resultsCache.autocompleteEnabled &&
+            this.s3CacheClient.isEnabled();
+
+        if (!forceRefresh && isCacheEnabled) {
+            const isCached = await this.s3CacheClient.getResultsMetadata(
```

Comment:
getResultsMetadata does not throw an error if results does not exist, so this needs to be called before getResults. It works like `hasResults`